lib/app_profiler/backend.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module AppProfiler
+  class Backend
+    def run(params = {}, &block)
+      raise NotImplementedError
+    end
+
+    def start(params = {})
+      raise NotImplementedError
+    end
+
+    def stop
```

Comment:
another one which I am 🤔 -- looks like we will always use stop followed by results.